### PR TITLE
EPMRPP-37186 React. Adjust Email notification component

### PR DESCRIPTION
**Pull request by AmsterGet**

### CODE REVIEW

File: app/src/pages/inside/settingsPage/settingsTabs/notificationsTab/forms/notificationRuleForm/validation.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    ruleOne.launchNameRule.sort().every((launch) => ruleTwo.launchNameRule.includes(launch)) &&
+    ruleOne.recipients.sort().every((recipient) => ruleTwo.recipients.includes(recipient))
+  );
+};
+
+export const validate = ({ rules }) => {
+  const errors = {};
+  const rulesArrayErrors = [];
+
+  rules.forEach((item, index, arr) => {
```
Comment (DiscoElevator) [inline app/src/pages/inside/settingsPage/settingsTabs/notificationsTab/forms/notificationRuleForm/validation.js]: This function is difficult to understand (see comments below). I'm not sure, but will my version be correct too?
```
const rulesErrors = rules.map((rule, index) => {
  if (rules.length > 1) {
    const hasDuplicates = arr.find((item, i) => compareRules(item, rule) && i !== index);
    if (hasDuplicates) {
      return { confirmed: 'hasDuplicates' };
    }
  }
  if (rule.recipients.length === 0 && !rule.informOwner) {
    return {
      recipients: 'recipientsHint',
      confirmed: 'error',
    };
  }
  return null;
}).filter(Boolean);
```

File: app/src/pages/inside/settingsPage/settingsTabs/notificationsTab/notificationsTab.jsx [hunk 2]
```diff
@@ -19,33 +19,34 @@ const cx = classNames.bind(styles);
   (state) => ({
     projectRole: activeProjectRoleSelector(state),
     userRole: userAccountRoleSelector(state),
-    emailEnabled: projectEmailEnabledSelector(state),
-    emailCases: projectEmailCasesSelector(state),
+    enabled: notificationIntegrationSelector(state).enabled,
```
Comment (DiscoElevator) [inline app/src/pages/inside/settingsPage/settingsTabs/notificationsTab/notificationsTab.jsx]: Please add corresponding selector to incapsulate name and position of this property.

File: app/src/pages/inside/settingsPage/settingsTabs/notificationsTab/notificationsTab.jsx [hunk 3]
```diff
@@ -55,38 +56,51 @@ export class NotificationsTab extends Component {
...
+
   isAbleToEditForm = () =>
     canConfigreEmailNotifications(this.props.userRole, this.props.projectRole);
-  isAbleToSubmit = ({ emailCases }) => emailCases.every(({ confirmed }) => confirmed);
-  convertEmailCaseForSubmission = (obj) => {
-    const { informOwner, launchNames, recipients, sendCase, tags } = obj;
+
+  isAbleToSubmit = ({ rules }) => rules.every(({ confirmed }) => confirmed);
+
+  convertRulesForSubmission = (obj) => {
```
Comment (DiscoElevator) [inline app/src/pages/inside/settingsPage/settingsTabs/notificationsTab/notificationsTab.jsx]: `convertRuleForSubmission ` - looks like this function converts only one rule

File: app/src/pages/inside/settingsPage/settingsTabs/notificationsTab/forms/notificationRuleForm/validation.js (hunk 1 above)
Comment (AmsterGet) [reply app/src/pages/inside/settingsPage/settingsTabs/notificationsTab/forms/notificationRuleForm/validation.js]: But these errors in different if blocks can be found at the same time. So, I use your solution with a few updates.

File: app/src/pages/inside/settingsPage/settingsTabs/notificationsTab/notificationsTab.jsx (hunk 3 above)
Comment (AmsterGet) [reply app/src/pages/inside/settingsPage/settingsTabs/notificationsTab/notificationsTab.jsx]: Fixed

File: app/src/pages/inside/settingsPage/settingsTabs/notificationsTab/forms/notificationRuleForm/validation.js (hunk 1 above)
Comment (AmsterGet) [reply app/src/pages/inside/settingsPage/settingsTabs/notificationsTab/forms/notificationRuleForm/validation.js]: I just change the order of these conditions according to their priority

**Pull request merged**